example/src/screens/CreateRoom.tsx

```diff
@@ -0,0 +1,267 @@
+import { BackgroundWrapper } from '@components/BackgroundWrapper';
+import { Modal } from '@components/Modal';
+import { TextInput } from '@components/TextInput';
+import { Typo } from '@components/Typo';
+import { StandardButton } from '@components/buttons/StandardButton';
+import { SERVER_URL } from '@env';
+import * as Membrane from '@jellyfish-dev/react-native-membrane-webrtc';
+import { RootStack } from '@model/NavigationTypes';
+import { useHeaderHeight } from '@react-navigation/elements';
+import type { NativeStackScreenProps } from '@react-navigation/native-stack';
+import React, { useEffect, useState, useCallback } from 'react';
+import {
+  Alert,
+  View,
+  StyleSheet,
+  KeyboardAvoidingView,
+  Platform,
+  PermissionsAndroid,
+  ScrollView,
+} from 'react-native';
+import { useVideoroomState } from 'src/VideoroomContext';
+
+function isEmpty(value) {
+  return value == null || value.trim().length === 0;
+}
+
+type Props = NativeStackScreenProps<RootStack, 'CreateRoom'>;
+type GoBackAction = Readonly<{
+  type: string;
+  payload?: object | undefined;
+  source?: string | undefined;
+  target?: string | undefined;
+}>;
+
+export const CreateRoom = ({ navigation, route }: Props) => {
+  const height = useHeaderHeight();
+  const [username, setUsername] = useState('');
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [modalAction, setModalAction] = useState<GoBackAction>();
+  const { roomName, setRoomName } = useVideoroomState();
+  const isSimulcastOn = true;
+
+  const { connect: mbConnect, joinRoom, error } = Membrane.useMembraneServer();
+
+  const params = {
+    token: 'NOW_YOU_CAN_SEND_PARAMS',
+  };
+
+  useEffect(() => {
+    if (error) {
+      console.log(error);
+      Alert.alert('Error when connecting to server:', error);
+    }
+  }, [error]);
+
+  useEffect(() => {
+    if (route.params?.roomName) {
+      setRoomName(route.params?.roomName);
+    }
+  }, []);
+
+  useEffect(
+    () =>
+      navigation.addListener('beforeRemove', (e) => {
+        if (!roomName && !username) {
+          // If we don't have unsaved changes, then we don't need to do anything
+          return;
+        }
+        e.preventDefault();
+        setModalAction(e.data.action);
```

Comment:
you don't have to use state for this modal action:
- if you need to re-render components because of a change, use `useState` - for example we want to use `useState` for room name because we want the change to be visible in the ui in the text box
- if you don't need to re-render anything but just want to save a variable then use `useRef` -it's better for performance (no re-renders!) and easier to use (they're synchronous - so if you set a ref it will be set immediately like a normal variable)

Reply:
Changed to use `useRef`.